client/components/ToolBar.tsx

```diff
@@ -19,36 +23,79 @@ export default function ToolBar({ projects, setSelectedProject, projectLoading }
+                    }}
+                />
+            );
+            break;
+        }
+        case 'SETTINGS': {
+            selectedPane = (
+                <SettingsPane
+                    // TODO: Move this into state and get it from the db
+                    rateLimiterConfig={rateLimiterConfig}
+                    rateLimiterLoading={rateLimiterLoading}
+                    setRateLimiterConfig={setRateLimiterConfig}
+                />
+            );
+            break;
+        }
+        default: {
+            selectedPane = null;
+        }
+    }
+
     return (
-        <div id="toolBar" style={extended === true ? { left: '0px' } : { left: '-340px' }}>
-            {/** //ToDo: make the tool bar look nice. 
+        <div id="toolBar" className={`toolBar ${!showToolbar ? 'closed' : ''}`}>
+            {/** //TODO: make the tool bar look nice. 
                    -toggle in and out from the right hand side. 
                    - sylethe project buttons */}
-
             <button
-                className="fa-solid fa-bars fa-2xl"
+                className={`arrow ${showToolbar ? 'left' : 'right'}`}
                 type="button"
                 aria-label="switch"
-                onClick={() => setExtended(extended !== true)}
+                onClick={() => {
+                    if (showToolbar && toolbarContent === 'PROJECTS') {
+                        // panel is open to projects. Close the panel
+                        setShowToolbar(false);
+                        setToolbarContent('');
+                    } else {
+                        // panel was either closed or on the settings view
+                        setShowToolbar(true);
+                        setToolbarContent('PROJECTS');
+                    }
+                }}
             />
-            <div id="toolBarMenu">
-                Projects{' '}
-                {projectLoading ? (
-                    <Loading />
-                ) : (
-                    <>
-                        {projects?.map((project) => (
-                            <ProjectItem
-                                project={project}
-                                setSelectedProject={setSelectedProject}
-                                setExtended={setExtended}
-                            />
-                        ))}
-                        <button type="button">Create a new Project</button>
-                        <div>Menu toSee/update profile info</div>
-                    </>
-                )}
+            <div id="toolBarButtons">
+                <button
+                    className="fa-solid fa-gear fa-2xl fa-bars"
+                    type="button"
+                    aria-label="switch"
+                    onClick={() => {
+                        if (showToolbar && toolbarContent === 'SETTINGS') {
+                            // panel is open to projects. Close the panel
+                            setShowToolbar(false);
```

Comment:
I think this should be
- if your viewing settings, show projects. So you can toggle between projects and settings with this button.
- I don't think this button should close the tool bar, leave that functionality up to the arrow button

Reply:
Thanks, needed some feedback on this